pkg/api/v1/pod/util.go

```diff
@@ -248,6 +252,22 @@ func IsPodReadyConditionTrue(status v1.PodStatus) bool {
 	return condition != nil && condition.Status == v1.ConditionTrue
 }
 
+// HasPodBeenReadyBefore returns true if the pod is ready or if it's not ready
+func HasPodBeenReadyBefore(pod *v1.Pod) bool {
+	_, readyCondition := GetPodCondition(&pod.Status, v1.PodReady)
+	if readyCondition == nil {
+		return false
+	}
+	if readyCondition.Status == v1.ConditionTrue {
+		return true
+	}
+	lastReady := readyCondition.LastTransitionTime.Time
+	created := pod.CreationTimestamp.Time
+	// If last status change was longer than maxDelayOfInitialReadinessStatus after the pod was
+	// created assume it was ready in the past.
+	return lastReady.After(created.Add(maxDelayOfInitialReadinessStatus))
```

Comment:
This logic assumes pod is scheduled within 10s which is not always true. If there is no node readily available pod may be pending for arbitrary amount of time. I think we should use PodScheduled condition's transition time as a baseline here. But that's just an educated guess - I think you should test the timing in different scenarios.

Reply:
We should probably be using `.status.startTime`

Reply:
Yup, now that I learned a bit more about pod lifecycle that makes more sense. Added a commit addressing that.